### PR TITLE
Add GCS to supported storage options

### PR DESCRIPTION
**Pull request by Cortey**
**Description**

**Related issue(s)**
https://github.com/kyma-project/docker-registry/issues/65

### CODE REVIEW

**Review by pPrecel — COMMENTED** [review]
Please, run the `make -C components/operator generate` command to regenerate the `api/v1alpha1/zz_generated.deepcopy.go` file.

**Review by pPrecel — APPROVED** [review]

File: docs/user/technical-reference/04-30-storage-configuration.md [hunk 1]
```diff
@@ -95,3 +95,36 @@ data:
   accessKey: "YWNjZXNzS2V5"
   secretKey: "c2VjcmV0S2V5"
 ```
+
+## GCS
```
Comment (IwonaLanger) [inline docs/user/technical-reference/04-30-storage-configuration.md:99]:
```suggestion
## Google Cloud Storage
```

File: docs/user/technical-reference/04-30-storage-configuration.md [hunk 2]
```diff
@@ -95,3 +95,36 @@ data:
...
+
+Google cloud storage can be configured using the `spec.storage.gcs` field. The only required field is `bucket` which contains the GCS bucket name. This storage type allows you to provide additional optional configuration, which is described in [DockerRegistry CR](../resources/06-20-docker-registry-cr.md). One of the optional configurations is the `secretName` that contains the authentication method to the GCS storage which is a private service account key in JSON format.
```
Comment (IwonaLanger) [inline docs/user/technical-reference/04-30-storage-configuration.md:101]:
```suggestion
Google Cloud Storage (GCS) can be configured using the **spec.storage.gcs** field. The only required field is the **bucket**, which contains the GCS bucket name. This storage type allows you to provide additional optional configuration described in [DockerRegistry CR](../resources/06-20-docker-registry-cr.md). One of the optional configurations is the **secretName**, which contains the authentication method to the GCS storage, which is a private service account key in the JSON format.
```

Comment (IwonaLanger) [reply docs/user/technical-reference/04-30-storage-configuration.md:101]: If GCS stands for Google Cloud Storage, does it make sense to say `GCS storage`?

File: docs/user/technical-reference/04-30-storage-configuration.md [hunk 3]
```diff
@@ -95,3 +95,36 @@ data:
...
+
+### Sample CR
```
Comment (IwonaLanger) [inline docs/user/technical-reference/04-30-storage-configuration.md:103]:
```suggestion
### Sample Custom Resource
```

**Review by anoipm — DISMISSED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Review by anoipm — APPROVED** [review]

**Review by anoipm — APPROVED** [review]

**Pull request merged**